pubsub/aws/snssqs/metadata.go

```diff
@@ -0,0 +1,252 @@
+	Region string
+	// name of the queue for this application. The is provided by the runtime as "consumerID".
+	sqsQueueName string
+	// name of the dead letter queue for this application.
+	sqsDeadLettersQueueName string
+	// flag to SNS and SQS FIFO.
+	fifo bool
+	// a namespace for SNS SQS FIFO to order messages within that group. limits consumer concurrency if set but guarantees that all
+	// published messages would be ordered by their arrival time to SQS.
+	// see: https://aws.amazon.com/blogs/compute/solving-complex-ordering-challenges-with-amazon-sqs-fifo-queues/
+	fifoMessageGroupID string
+	// amount of time in seconds that a message is hidden from receive requests after it is sent to a subscriber. Default: 10.
+	messageVisibilityTimeout int64
+	// number of times to resend a message after processing of that message fails before removing that message from the queue. Default: 10.
+	messageRetryLimit int64
+	// upon reaching the messageRetryLimit, disables the default deletion behaviour of the message from the SQS queue, and resetting the message visibilty on SQS
+	// so that other consumers can try consuming that message.
+	disableDeleteOnRetryLimit bool
+	// if sqsDeadLettersQueueName is set to a value, then the messageReceiveLimit defines the number of times a message is received
+	// before it is moved to the dead-letters queue. This value must be smaller than messageRetryLimit.
+	messageReceiveLimit int64
+	// amount of time to await receipt of a message before making another request. Default: 1.
+	messageWaitTimeSeconds int64
+	// maximum number of messages to receive from the queue at a time. Default: 10, Maximum: 10.
+	messageMaxNumber int64
+	// disable resource provisioning of SNS and SQS.
+	disableEntityManagement bool
+	// assets creation timeout.
+	assetsManagementTimeoutSeconds float64
+	// aws account ID. internally resolved if not given.
+	accountID string
+}
+
+func getAliasedProperty(aliases []string, metadata pubsub.Metadata) (string, bool) {
+	props := metadata.Properties
+	for _, s := range aliases {
+		if val, ok := props[s]; ok {
+			return val, true
+		}
+	}
+
+	return "", false
+}
+
+func parseInt64(input string, propertyName string) (int64, error) {
+	number, err := strconv.Atoi(input)
+	if err != nil {
+		return -1, fmt.Errorf("parsing %s failed with: %w", propertyName, err)
+	}
+
+	return int64(number), nil
+}
+
+func parseBool(input string, propertyName string) (bool, error) {
+	val, err := strconv.ParseBool(input)
+	if err != nil {
+		return false, fmt.Errorf("parsing %s failed with: %w", propertyName, err)
+	}
+	return val, nil
+}
+
+func parseFloat64(input string, propertyName string) (float64, error) {
+	val, err := strconv.ParseFloat(input, 64)
+	if err != nil {
+		return 0, fmt.Errorf("parsing %s failed with: %w", propertyName, err)
+	}
+	return val, nil
+}
+
+//nolint:cyclop
+func (s *snsSqs) getSnsSqsMetatdata(metadata pubsub.Metadata) (*snsSqsMetadata, error) {
```

Comment:
This function is really long as I do not think of a good idea to solve it.

/cc @artursouza do you have a good idea for this?

Reply:
done. I have broken it up